build.go

```diff
@@ -172,50 +220,123 @@ func runTestCommandLine(pkg string, tags []string) []string {
 }
 
-func (tr *testRunner) Name() string {
+var _ mg.Fn = &TestRunner{}
+
+// Name implements [mg.Fn].
+func (tr *TestRunner) Name() string {
 	return tr.ID()
 }
 
-func (tr *testRunner) ID() string {
+// ID implements [mg.Fn].
+func (tr *TestRunner) ID() string {
 	return fmt.Sprintf("run-test-%s", tr.pkg)
 }
 
-func (tr *testRunner) Run(ctx context.Context) error {
-	return RunTest(ctx, tr.pkg, tr.tags...)
+// Run implements [mg.Fn]. It runs the package's test with "go test."
+func (tr *TestRunner) Run(ctx context.Context) error {
+	mg.CtxDeps(ctx, BuildTest(tr.pkg, tr.tags...))
+
+	return sh.RunV(mg.GoCmd(), runTestCommandLine(tr.pkg, tr.tags)...)
 }
 
-// RunTestDep returns a [mg.Fn] that will run the tests for the given package, subject to the given build tags.
-func RunTestDep(pkg string, tags ...string) mg.Fn {
-	return &testRunner{pkg, tags}
+// RunTest returns a [mg.Fn] that will run the tests for the given package, subject to the given build tags.
+func RunTest(pkg string, tags ...string) *TestRunner {
+	return &TestRunner{pkg, tags}
 }
 
-// Test runs unit tests.
-func Test(ctx context.Context, tags ...string) error {
-	// It's technically not necessary to build the tests before running them; "go test" will build them anyway.
+// AllGinkgoTestRunner is a [mg.Fn] that identifies all tests in the project and uses Ginkgo to build and run them.
+type AllGinkgoTestRunner struct {
+	AllTestRunner
+	bin string
+}
+
+var _ mg.Fn = &AllGinkgoTestRunner{}
+
+// Run implements [mg.Fn]. It uses Ginkgo to build test binaries for all applicable packages in the project (as by
+// [AllGinkgoTestBuilder], and then uses Ginkgo to run them all.
+func (agtr *AllGinkgoTestRunner) Run(ctx context.Context) error {
+	// It's technically not necessary to build the tests before running them; "ginkgo run" would build them anyway.
 	// However, we specify BuildTests as a dependency so that _all_ the tests get built before _any_ of them start
 	// running. That makes the output cleaner because lengthy test output doesn't push any build failures off the
 	// top of the screen.
-	mg.CtxDeps(ctx, LoadDependencies, BuildTestsDep(tags...))
+	mg.CtxDeps(ctx,
+		LoadDependencies,
+		Install(agtr.bin, "github.com/onsi/ginkgo/v2/ginkgo"),
+		BuildTests(agtr.tags...).UseGinkgo(agtr.bin),
+	)
+	args := []string{
+		"run",
+		"-p",
+		"--timeout", "10s",
+	}
+	args = append(args, formatTags(ginkgoTagOpt, agtr.tags)...)
+	for _, info := range Packages {
+		if !info.HasTest() {
+			continue
+		}
+		args = append(args, info.TestBinary())
+	}
+	return sh.Run(agtr.bin, args...)
+}
+
+// AllTestRunner implements [mg.Fn] to identify, build, and run tests for all packages in the current project.
+type AllTestRunner struct {
+	tags []string
+}
+
+var _ mg.Fn = &AllTestRunner{}
+
+// Name implements [mg.Fn].
+func (atr *AllTestRunner) Name() string {
+	return atr.ID()
+}
+
+// ID implements [mg.Fn].
+func (*AllTestRunner) ID() string {
+	return fmt.Sprintf("run-all-tests")
```

Comment:
We don't really need `Sprintf` for this string literal.
```suggestion
	return "run-all-tests"
```